RN, initial version (#440)
percona/percona-xtradb-cluster-operator: source/ReleaseNotes/Kubernetes-Operator-for-PXC-RN1.5.0.rst

.. _K8SPXC-1.5.0:

================================================================================
*Percona Kubernetes Operator for Percona XtraDB Cluster* 1.5.0
================================================================================

:Date: July 21, 2020
:Installation: `Installing Percona Kubernetes Operator for Percona XtraDB Cluster <https://www.percona.com/doc/kubernetes-operator-for-psmongodb/index.html#installation>`_

New Features
================================================================================

* :jirabug:`K8SPXC-298`: Automatic synchronization of MySQL users with ProxySQL
* :jirabug:`K8SPXC-294`: HAProxy Support
* :jirabug:`K8SPXC-284`: Fully automated minor version updates
* :jirabug:`K8SPXC-257`: Update Reader members before Writer member at cluster upgrades
* :jirabug:`K8SPXC-256`: Support multiple PXC minor versions by the Operator

Improvements
================================================================================

* :jirabug:`K8SPXC-290`: Extend usable backup schedule syntax to include lists of values
* :jirabug:`K8SPXC-309`: Quickstart Guide on Google Kubernetes Engine (GKE)

> * :jirabug:`K8SPXC-288`: Quickstart Guide on Amazon Elastic Kubernetes Service (EKS)

Reviewer: links?

* :jirabug:`K8SPXC-279`: Use SYSTEM_USER privilege for system users on PXC 8.0

> * :jirabug:`K8SPXC-277`: Install GDB in all PXC images

Reviewer: ```suggestion
* :jirabug:`K8SPXC-277`: Install GDB in PXC images
```
too many ways to read it incorrectly

.. _K8SPXC-1.5.0:

================================================================================
*Percona Kubernetes Operator for Percona XtraDB Cluster* 1.5.0
================================================================================

> :Date: July 13, 2020

Reviewer: ```suggestion
:Date: July 21, 2020
```

:Installation: `Installing Percona Kubernetes Operator for Percona XtraDB Cluster <https://www.percona.com/doc/kubernetes-operator-for-psmongodb/index.html#installation>`_

New Features
================================================================================

> * :jirabug:`K8SPXC-298`: Automatic synchronization of ProxySQL users

Reviewer: ```suggestion
* :jirabug:`K8SPXC-298`: Automatic synchronization users from MySQL into ProxySQL
```

* :jirabug:`K8SPXC-294`: HAProxy Support
* :jirabug:`K8SPXC-284`: Fully automated minor version updates

> * :jirabug:`K8SPXC-257`: Update Reader members before Writer member when upgrading cluster with ProxySQL

Reviewer: ```suggestion
* :jirabug:`K8SPXC-257`: Update Reader members before Writer member when upgrading cluster
```

* :jirabug:`K8SPXC-256`: Support multiple PXC minor versions by the Operator

Improvements
================================================================================

* :jirabug:`K8SPXC-290`: Extend usable backup schedule syntax to include lists of values
* :jirabug:`K8SPXC-309`: Quickstart Guide on Google Kubernetes Engine (GKE)
* :jirabug:`K8SPXC-288`: Quickstart Guide on Amazon Elastic Kubernetes Service (EKS)
* :jirabug:`K8SPXC-279`: Use SYSTEM_USER privilege for system users on PXC 8.0
* :jirabug:`K8SPXC-277`: Install GDB in all PXC images

> * :jirabug:`K8SPXC-276`: Pod-0 always Writer in ProxySQL

Reviewer: ```suggestion
* :jirabug:`K8SPXC-276`: Pod-0 should be selected as Writer if possible
```

> * :jirabug:`K8SPXC-252`: Automatically manage system users for MySQL and ProxySQL with rotation

Reviewer: ```suggestion
* :jirabug:`K8SPXC-252`: Automatically manage system users for MySQL and ProxySQL on password rotation (via secret modification)
```

> * :jirabug:`CLOUD-535`: Standard reporting protocol for debugging and troubleshooting

Reviewer: please remove, it is internal research

* :jirabug:`CLOUD-556`: Kubernetes 1.17 added to the list of supported platforms
* :jirabug:`CLOUD-404`: Support of loadBalancerSourceRanges for LoadBalancer Services

Bugs Fixed
================================================================================

> * :jirabug:`K8SPXC-327`: SST fail when PXC Pod restarted in the middle of SST

Reviewer: ```suggestion
* :jirabug:`K8SPXC-327`: CrashloopBackOff if PXC 8.0 Pod restarted in the middle of SST
```

> * :jirabug:`K8SPXC-270`: Restore job wiping data from the original backup's cluster when restoring to another cluster in the same namespace

Reviewer: ```suggestion
* :jirabug:`K8SPXC-270`: Restore job wiping data from the original backup's cluster when restoring to another cluster in the same namespace (Thanks to user Sergey Kuzmichev for reporting this issue)
```
?

> * :jirabug:`K8SPXC-352`: Backup cronjob not scheduled (Thanks to user msavchenko for reporting this issue)

Reviewer: ```suggestion
* :jirabug:`K8SPXC-352`: Backup cronjob not scheduled in some Kubernetes environments (Thanks to user msavchenko for reporting this issue)
```

> Backup cronjob is not scheduled

Reviewer: duplicate
```suggestion

```

* :jirabug:`K8SPXC-275`: Outdated documentation on the Operator updates (Thanks to user martin.atroo for reporting this issue)

> * :jirabug:`K8SPXC-347`: XtraBackup fail after uploading a backup, causing the backup process restart (Thanks to user connde for reporting this issue)

Reviewer: ```suggestion
* :jirabug:`K8SPXC-347`: XtraBackup fail after uploading a backup, causing the backup process restart in some cases (Thanks to user connde for reporting this issue)
```

> * :jirabug:`K8SPXC-331`: pxc-entrypoint.sh: no such file or directory error when running 5.7
> * :jirabug:`K8SPXC-330`: missing online nodes in reader hostgroup while upgrade in progress

Reviewer: please remove, both created during 1.5.0 development
```suggestion
```

> * :jirabug:`K8SPXC-326`: Pod recreated during PXC Pod downsizing

Reviewer: I have reworked wording inside the ticket

> * :jirabug:`K8SPXC-242`: Backup script running indefinitely on SST startup error

Reviewer: I have reworked wording inside the ticket, not it is improvement

Author: How can we name this improvement so that the title would be descriptive? "Stability improvements in the internal backup implementation for PXC 8.0"? Just to explain the essence of this improvement in the title.

Reviewer: I have updated ticket title to `Improve internal backup implementation for PXC 8.0`

> * :jirabug:`K8SPXC-230`: Backup fail if just one PXC instance running

Reviewer: I have removed 1.5.0 from the ticket, it is strange, no commits made

* :jirabug:`K8SPXC-323`: Missing ``tar`` utility in the PXC node docker image
* :jirabug:`CLOUD-474`: Cluster creation not failing if wrong resources are set
* :jirabug:`CLOUD-531`: Wrong usage of ``strings.TrimLeft`` when processing apiVersion

> * :jirabug:`PXC-2987`: SST incompatible between 5.7 and 8.0

Reviewer: I have removed 1.5.0 from the ticket, it is strange, no commits made
